ec2-worker/asg.tf

```diff
@@ -16,48 +16,51 @@ data "aws_ami" "ubuntu" {
 }
 
 module "is_ebs_optimised" {
-  source        = "github.com/skyscrapers/terraform-instances//is_ebs_optimised?ref=2.3.8"
-  instance_type = "${var.instance_type}"
+  source        = "github.com/skyscrapers/terraform-instances//is_ebs_optimised?ref=3.0.1"
+  instance_type = var.instance_type
 }
 
 resource "aws_launch_template" "concourse_worker_launchtemplate" {
-  count         = "${var.work_disk_ephemeral ? 0 : 1}"
-  image_id      = "${length(var.custom_ami) == 0 ? data.aws_ami.ubuntu.id : var.custom_ami}"
-  instance_type = "${var.instance_type}"
-  key_name      = "${var.ssh_key_name}"
-  user_data     = "${data.template_cloudinit_config.concourse_bootstrap.rendered}"
-  ebs_optimized = "${module.is_ebs_optimised.is_ebs_optimised}"
+  count         = var.work_disk_ephemeral ? 0 : 1
+  image_id      = length(var.custom_ami) == 0 ? data.aws_ami.ubuntu.id : var.custom_ami
```

Comment:
We should use a `null` here instead of an empty string as default value for `custom_ami`.

Then you would use it like `image_id = coalesce(var.custom_ami, data.aws_ami.ubuntu.id)`